### DISCUSSION TITLE
How to render conditional table rows ?

### DISCUSSION THREAD

**Discussion by martin-g** [quarto-dev/quarto-cli#7795] [Q&A] [answered]
### Description

I need to render some table rows only when a specific Quarto profile is activated. 

Here is a simple reproducer:

```qmd

|    Variable    |      Meaning     |
|----------------|------------------|
| var1  | some value 1 |
| var2  | some value 2 |
::: {.content-visible when-profile="dummy"}
| var3  | some value 3 |
| var4  | some value 4 |
:::
```

I render it with: `quarto render --profile dummy tables.qmd --to pdf ; open tables.pdf`

But the rendered table is broken because of the empty line due to `::: {.content-visible when-profile="dummy"}`:

![quarto-table-row-profile](https://github.com/quarto-dev/quarto-cli/assets/232002/b7530418-8701-44cb-9fe5-0c87a64b4170)

What is the correct syntax ?

Thank you!

**Comment by mcanouil** [comment]
You can't make half tables and never will be able to do so.

Either generate the table programmatically and use the fact that profiles are defined using environment variables or duplicate part of the table.

````qmd
::: {.content-visible unless-profile="dummy"}
|    Variable    |      Meaning     |
|----------------|------------------|
| var1  | some value 1 |
| var2  | some value 2 |
:::

::: {.content-visible when-profile="dummy"}
|    Variable    |      Meaning     |
|----------------|------------------|
| var1  | some value 1 |
| var2  | some value 2 |
| var3  | some value 3 |
| var4  | some value 4 |
:::
````

**Reply by martin-g** [reply]
Thanks for the quick answer, @mcanouil !
Could you please point me to an example of a programmatically generated table/content ?

**Reply by mcanouil** [reply, marked as answer]
I am not sure to understand.
it's just regular computation with whatever language you want to use.
There is a whole set of guides regarding computations, <https://quarto.org/docs/guide/>.
And regarding `QUARTO_PROFILE`, it is in the documentation you likely have read since you are using profiles, see <https://quarto.org/docs/projects/profiles.html#activating-profiles>.
````qmd
```r
df <- data.frame(
  Variable = c("var1", "var2", "var3", "var4"),
  Meaning = c("some value 1", "some value 2", "some value 3", "some value 4")
)

if (Sys.getenv("QUARTO_PROFILE") %in% "dummy") {
  knitr::kable(df)
} else {
  knitr::kable(df[1:2, ])
}
```
````

**Reply by martin-g** [reply]
Thank you!

Actually, I am not using any languages (yet)!
I am working on a proof of concept for reworking OpenHPC's LaTeX based documentation to something more friendlier (like Markdown) that supports localization (I use Quarto's profiles for this).

**Reply by mcanouil** [reply]
I see, then it might be better to not use computation, as you will need to deal with dependencies (R or Python).

**Reply by jimjam-slam** [reply]
Since `.content-visible` uses the Pandoc div syntax, there's a good chance it can't be spliced into the middle of a table (that's me speculating!)

But if you don't want to deal with computational dependencies, you could put two versions of the whole table in `.content-visible`, the downside being some duplication :)

**Reply by mcanouil** [reply]
That's https://github.com/quarto-dev/quarto-cli/discussions/7795#discussioncomment-7764261 😉

**Reply by jimjam-slam** [reply]
Haha oops!
